obs-packaging/gen_versions_txt.sh

```diff
@@ -148,13 +150,57 @@ main() {
 			use_head=1
 			shift
 			;;
+		--tag)
+			use_tag=1
+			shift
+			;;
 		-*)
 			die "Invalid option: ${1:-}" "1"
 			shift
 			;;
 	esac
-	local branch="${1:-}"
-	[ -n "${branch}" ] || die "No branch specified" "1"
+
+	local kata_version=
+	if [ -n "$use_tag" ]; then
+		if [ -n "${use_head}" ]; then
+		       die "tag and head options are mutually exclusive"
+		fi
+
+		# We are generating versions based on the provided tag
+		local tag="${1:-}"
+		[ -n "${tag}" ] || die "No tag specified" "1"
+
+		# use the runtime's repository to determine branch information
+		local repo="github.com/kata-containers/runtime"
+		local repo_dir="runtime"
+		git clone --quiet "https://${repo}.git" "${repo_dir}"
+		pushd "${repo_dir}" >> /dev/null
+		local branch=$(git branch -r -q --contains "${tag}" | grep -E "master|stable" | grep -v HEAD)
+
+		popd >> /dev/null
+		rm -rf ${repo_dir}
+
+		[ -n "${branch}" ] || die "branch for tag ${tag} not found"
+
+		# in the event this is on master as well as stable, or multiple stables, just pick the first branch
+		# (ie, 1.8.0-alpha0 may live on stable-1.8 as well as master: we'd just use master in this case)
+		branch=$(echo ${branch} | awk -F" " '{print $1}')
```

Comment:
if there are two branches the case I think about is because we just move from alpha to rc-0, in this, I'd say we continue the release process in master.  In any other case with to branches we pick the stable branch from to the release.